### DISCUSSION TITLE
Chrome 92 slow to render

### DISCUSSION THREAD

**Discussion by pms1969** [concourse/concourse#7410] [Help & Support]
We've got issues with rendering pipelines in Chrome 92?  We have noticed a significant performance degradation since the update from 91. Concourse version 7.1.0
Is this a known issue?  I've not been able to find anything related to it.

<img width="394" alt="image" src="https://user-images.githubusercontent.com/2762128/129026949-0f298184-3f63-4148-a6d1-e827cfdff1e6.png">
As the above image shows, most of the time is in rendering.

It isn't a problem with Firefox, and isn't a problem with Safari.  Any chromium based browsers however exhibit the same behaviour.

**Comment by aoldershaw** [comment]
Maybe the same as https://github.com/concourse/concourse/issues/7329?

I don't think anything changed here in Concourse itself recently, interesting that you've possibly pinpointed it to Chrome 92.

If it is the same as the linked issue, I think we just have to use a more sane animation than rendering multiple box shadows for the running jobs "ripple" effect: https://github.com/concourse/concourse/blob/34c5127758908806f5598995c842a196758b0ed7/web/assets/css/colors.less#L123-L154

**Comment by pms1969** [comment]
I'd like to think that it is the animations, but I took that test when there were no jobs running.  With jobs running, it is significantly slower (30+ seconds to render).  Even turning off the animation (as I have permanently done via a browser plugin), it's still painfully slow.

To be fair, the one pipeline that this is a particular problem on is a monster.  There are 28 jobs and 40+ resources that all need rendering.  We might just be pushing the limits of what is sane for the JS renderer.

**Reply by aoldershaw** [reply]
I took a look at how we could potentially optimize the initial render of the SVG, and found some low relatively hanging fruit that should hopefully improve things: https://github.com/concourse/concourse/pull/7438

It reduced the render time of https://ci.concourse-ci.org/teams/main/pipelines/concourse?group=all from ~750ms to ~100ms on my machine (running Chrome 92)

**Reply by pms1969** [reply]
latest version of Chrome seems to have fixed the performance issues.
